src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala

```diff
@@ -121,157 +120,166 @@ class SvPileup
+        // Special case for 2 since most templates will generate two segments and we'd like it to be efficient
+        val bp = findBreakpoint(segments.head, segments.last, maxWithinReadDistance, maxReadPairInnerDistance)
+        if (bp.isEmpty) NoBreakpoints else bp.toIndexedSeq
+      case _     =>
+        val builder = IndexedSeq.newBuilder[BreakpointEvidence]
+        segments.iterator.sliding(2).foreach { case Seq(seg1, seg2) =>
+          findBreakpoint(seg1, seg2, maxWithinReadDistance, maxReadPairInnerDistance).foreach(builder += _)
+        }
+        builder.result()
+    }
+  }
 
-    if (segments.length == 1) IndexedSeq.empty else {
-      segments.iterator.sliding(2).flatMap { case Seq(seg1, seg2) =>
-        var result                 = determineInterContigBreakpoint(seg1, seg2)
-        if (result.isEmpty) result = determineIntraContigBreakpoint(seg1, seg2, maxAlignedSegmentInnerDistance , maxReadPairInnerDistance)
-        if (result.isEmpty) result = determineOddPairOrientation(seg1, seg2)
-        result
-      }
-      .toIndexedSeq
+  /** Checks to see if there is a breakpoint between two segments and returns it, or None. */
+  private def findBreakpoint(seg1: AlignedSegment,
+                             seg2: AlignedSegment,
+                             maxWithinReadDistance: Int,
+                             maxReadPairInnerDistance: Int): Option[BreakpointEvidence] = {
+    if (isInterContigBreakpoint(seg1, seg2) ||
+        isIntraContigBreakpoint(seg1, seg2, maxWithinReadDistance, maxReadPairInnerDistance)
+    ) {
+      val bp = Breakpoint(seg1, seg2)
+      val ev = if (seg1.origin.isInterRead(seg2.origin)) EvidenceType.ReadPair else EvidenceType.SplitRead
+      Some(BreakpointEvidence(bp, ev))
+    }
+    else {
+      None
     }
   }
 
-  /** Determines if two segments are evidence of an SV joining two different contigs.
+  /** Determines if two segments are evidence of breakpoint joining two different contigs.
    *
    * @param seg1 the first alignment segment
    * @param seg2 the second alignment segment
    */
-  def determineInterContigBreakpoint(seg1: AlignedSegment, seg2: AlignedSegment): Option[BreakpointEvidence] = {
+  def isInterContigBreakpoint(seg1: AlignedSegment, seg2: AlignedSegment): Boolean = {
     val r1 = seg1.range
     val r2 = seg2.range
-    if (r1.refIndex == r2.refIndex) None
-    else {
-      val evidence = if (seg1.origin.isPairedWith(seg2.origin)) ReadPairInterContig else SplitReadInterContig
-      Some(BreakpointEvidence(from=seg1, into=seg2, evidence=evidence))
-    }
+    r1.refIndex != r2.refIndex
   }
 
-  /** Determines if the primary mappings are from a read pair are evidence of an SV joining two different regions from
-   * the same contig.
+  /** Determines if the two segments are provide evidence of a breakpoint joining two different regions from
+   * the same contig. Returns true if:
+   *   - the two segments overlap (implying some kind of duplication)
```

Comment:
I am not sure I agree with this by default.  What about short inserts?  How do we control this?

Reply:
I'll add a test, but in that case the two reads will generate two segments, which overlap and will get merged into a single large segment, so we won't see two overlapping segments.

Reply:
Added test, works as expected, added a comment here too.